.config/nvim/ftplugin/java.lua

```diff
@@ -1,142 +1,159 @@
 -- JDTLS (Java LSP) configuration
-local jdtls = require('jdtls')
-local project_name = vim.fn.fnamemodify(vim.fn.getcwd(), ':p:h:t')
-local workspace_dir = vim.env.HOME .. '/jdtls-workspace/' .. project_name
+local home = vim.env.HOME -- Get the home directory
+
+local jdtls = require("jdtls")
+local project_name = vim.fn.fnamemodify(vim.fn.getcwd(), ":p:h:t")
+local workspace_dir = home .. "/jdtls-workspace/" .. project_name
+
+-- Determine OS
+-- Use this option to avoid explicitly adding manually everytime you switch OS e.g "config_linux", "config_mac"
+-- This saves a lot of time if you are working with multiple OS e.g Linux or Mac OS
+if vim.fn.has("mac") == 1 then
+	CONFIG = "mac"
+elseif vim.fn.has("unix") == 1 then
+	CONFIG = "linux"
+else
+	print("Unsupported system")
```

Comment:
When the system is not found, can you change the message to:  OS not found, defaulting to 'linux'
and then set the variable to linux?

Reply:
Done modifying and including options for windows